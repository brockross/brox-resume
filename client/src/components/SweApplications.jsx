import React from 'react';

const SweApplications = (props) => {
  return (
    <div>
      <h3>Software Engineering Applications</h3>
      <ul>
        <li>Bandland</li>
        <li>Cinetix</li>
        <li>Jerry the Giraffe</li>
      </ul>
    </div>
  )
}

export default SweApplications;