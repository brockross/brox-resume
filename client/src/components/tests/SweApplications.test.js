import React from 'react';
import { shallow } from 'enzyme';

import SweApplications from  '../SweApplications.jsx';
import sampleData from '../../dummyData.js';

describe('SweApplications', () => {
  it('should render correctly', () => {
    const component = shallow(<SweApplications sweApps={JSON.parse(sampleData)}/>);

    expect(component).toMatchSnapshot();
  });
});