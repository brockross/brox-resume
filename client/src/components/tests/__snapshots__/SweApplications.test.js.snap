// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SweApplications should render correctly 1`] = `
<div>
  <h3>
    Software Engineering Applications
  </h3>
  <div>
    <h4>
      Bandland
       - 
      software engineer
    </h4>
    <p>
      PostgreSQL | Node | AWS
    </p>
    <p>
      <em>
        Legacy back-end codebase of app used for streaming albums, along with displaying album art, artist info, comments, and recommendations
      </em>
    </p>
    <ul>
      <li>
        Scaled a legacy database and Express API to handle web scale traffic (~10M database entries; 100 RPS)
      </li>
      <li>
        Improved request latency by 100% by load balancing with nginx
      </li>
      <li>
        Optimized static asset compression and delivery, taking Google PageSpeed score from 6 to 100
      </li>
      <li>
        Utilized agile workflow to complete individual and team deliverables, meeting or exceeding deadlines and quality standards
      </li>
    </ul>
  </div>
  <div>
    <h4>
      Cinetix
       - 
      software engineer
    </h4>
    <p>
      React | Node | MongoDB
    </p>
    <p>
      <em>
        Single-page app to display local showtimes, rating and synopsis info, associated videos, and cast/crew for a given movie
      </em>
    </p>
    <ul>
      <li>
        Designed and executed a React JS architecture with modularity & reusability as top priorities
      </li>
      <li>
        Built proxy server in Node JS to facilitate a service-oriented architecture; structured an API and service and deployed it to Amazon EBS
      </li>
      <li>
        Coordinated with a team of fellow engineers to ensure smooth service integration and met deadlines using agile workflow & frequent github PRs
      </li>
    </ul>
  </div>
  <div>
    <h4>
      Jerry the Giraffe
       - 
      software engineer
    </h4>
    <p>
      React | Node | AWS
    </p>
    <p>
      <em>
        Children’s book adapted to a ‘choose your own adventure’ app, featuring branching stories, multiple endings, and an inventory system
      </em>
    </p>
    <ul>
      <li>
        Implemented React Hooks for all state management
      </li>
      <li>
        Conceived and executed a modular ‘scene’ building system to facilitate cleanly organized decision trees
      </li>
      <li>
        Deployed via Express server on Amazon EC2 instance
      </li>
    </ul>
  </div>
</div>
`;
